### PR TITLE
Fixes #2674 - Add tmpdfiles.d configuration for systemd to rpm spec

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Untested, so give it a serious review ;)

### CODE REVIEW

File: extras/packaging/rpm/sources/foreman.tmpfiles [hunk 1]
```diff
@@ -0,0 +1 @@
+d /var/run/foreman 0755 foreman foreman -
```
Comment (domcleal) [inline extras/packaging/rpm/sources/foreman.tmpfiles]: Should just be /run on Fedora

File: foreman.spec [hunk 2]
```diff
@@ -385,6 +385,7 @@ install -d -m0750 %{buildroot}%{_localstatedir}/log/%{name}
 install -Dp -m0755 script/%{name}-debug %{buildroot}%{_sbindir}/%{name}-debug
 install -Dp -m0644 %{confdir}/%{name}.sysconfig %{buildroot}%{_sysconfdir}/sysconfig/%{name}
 install -Dp -m0755 %{confdir}/%{name}.init %{buildroot}%{_initrddir}/%{name}
+install -Dp -m0644 %{confdir}/%{name}.tmpfiles %{buildroot}%{_prefix}/lib/tmpfiles.d/%{name}.conf
```
Comment (domcleal) [inline foreman.spec]: Should only be installed on Fedora, see the conditionals in foreman-proxy.spec.

File: foreman.spec [hunk 3]
```diff
@@ -444,6 +445,7 @@ rm -rf %{buildroot}
 %config(noreplace) %{_sysconfdir}/sysconfig/%{name}
 %config(noreplace) %{_sysconfdir}/logrotate.d/%{name}
 %config %{_sysconfdir}/cron.d/%{name}
+%config %{_prefix}/lib/tmpfiles.d/%{name}.conf
```
Comment (domcleal) [inline foreman.spec]: Should only be on Fedora, shouldn't be a %config since it's in /usr/lib.

File: foreman.spec (hunk 2 above)
Comment (GregSutcliffe) [inline foreman.spec]: good point, i was copying from the ticket - i had forgotten they're symlinked these days ;)

**Comment by domcleal** [comment]
Missing a dependency on systemd of some sort for Fedora?

File: foreman.spec (hunk 2 above)
Comment (GregSutcliffe) [inline foreman.spec]: Oops, that was meant to be for the above comment

**Comment by GregSutcliffe** [comment]
@domcleal fixed, i think ;)

**Comment by GregSutcliffe** [comment]
@domcleal isn't systemd installed by default from f17 and up?

File: foreman.spec [hunk 4]
```diff
@@ -387,6 +387,9 @@ install -Dp -m0644 %{confdir}/%{name}.sysconfig %{buildroot}%{_sysconfdir}/sysco
 install -Dp -m0755 %{confdir}/%{name}.init %{buildroot}%{_initrddir}/%{name}
 install -Dp -m0644 %{confdir}/%{name}.logrotate %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
 install -Dp -m0644 %{confdir}/%{name}.cron.d %{buildroot}%{_sysconfdir}/cron.d/%{name}
+%if 0%{?fedora} > 16
```
Comment (domcleal) [inline foreman.spec]: For future-proofing, could you reuse the conditionals from foreman-proxy.spec which mean RHEL7 will also get these?

**Comment by domcleal** [comment]
Ah, I was thinking that the dependencies should reflect this.. but since we're not using it in a scriptlet, I think we're ok to leave them out.

**Comment by GregSutcliffe** [comment]
@domcleal I've inverted the logic in the foreman-proxy IFs since they put the systemd tmpfiles in an %else clause. Otherwise should be the same effect.

**Comment by domcleal** [comment]
scratch builds:
http://koji.katello.org/koji/taskinfo?taskID=40043 (F18, tested)
http://koji.katello.org/koji/taskinfo?taskID=40045 (EL6)

Merged as d30ac55, thanks @GregSutcliffe!

**Pull request closed without merging**